src/main/java/com/easytestit/generatexml/GenerateXML.java

```diff
@@ -1,145 +1,163 @@
     }
 
     /**
-     * Void function which provides the ability to create a directory that will later serve as the folder
+     * The method which provides the ability to create a directory that will later serve as the folder
      * where the aggregated XML report will be created
-     * </p>
+     *
      * Also, this folder will zip later if needed configuration will pass
      */
     private void createResultsReportFolder() {
-        LOGGER.info("Try to create new folder in project directory");
-        File resultsFolder = new File(DefaultData.REPORT_RESULTS_FOLDER);
-        LOGGER.info(!resultsFolder.exists() && resultsFolder.mkdirs() ? "Direction ".concat(DefaultData.REPORT_RESULTS_FOLDER).concat(" was created") : "Direction ".concat(DefaultData.REPORT_RESULTS_FOLDER).concat(" was already created"));
+        String path;
+        if (configuration == null) {
+            path = ConfigDataProvider.DEFAULT_FOLDER;
+        } else {
+            path = ConfigDataProvider.REPORT_RESULTS_FOLDER;
+        }
+
+        File resultsFolder = new File(path);
+        LOGGER.log(Level.INFO, !resultsFolder.exists() && resultsFolder.mkdirs()
+                ? "Direction " + path + " created successfully"
+                : "Direction " + path + " was already created"
+        );
     }
 
     /**
      * A convenience method which read files with specific extension
      * and collects path to these files in a new String collection
-     * </p>
-     * @param reportFolder the folder where locate all files after launching tests
+     *
+     * @param source the folder where locate all files after launching tests
      * @return the collection of String with defined path to JSON files
      */
-    @NotNull
-    private Collection<String> getJsonFilesFrom(@NotNull File reportFolder) {
-        var extensions = Collections.singletonList(DefaultData.DEFAULT_FILE_EXTENSIONS);
-        Collection<String> out = new ArrayList<>();
-        File[] files = reportFolder.listFiles();
-        //might be null
-        if (files == null) {
-            LOGGER.info("Report folder ".concat(reportFolder.getName()).concat(" aren't contain any files!"));
-            return out;
-        }
-        //analyze files in folder and filter by specified extension and collect these to new collection
-        Arrays.stream(files).forEach(file -> {
-            if (file.isFile() && extensions.contains(file.getName().substring(file.getName().lastIndexOf('.') + 1))) {
-                out.add(String.valueOf(reportFolder)
-                        .concat(String.valueOf(File.separatorChar))
-                        .concat(file.getName()));
-            } else {
-                LOGGER.info("Report folder ".concat(reportFolder.getName())
-                        .concat(" doesn't contain files with JSON format!"));
+    private Collection<String> resolvePath(final File source) {
+        Collection<String> filePathList = new ArrayList<>();
+        if (source != null) {
```

Comment:
```
if (source == null) {
     throw new GenerateXMLReportException("No file provided, but required");
}
```

or for the same cases you can create MissingRequiredArgumentException
```
if (source == null) {
     throw new MissingRequiredArgumentException("source");
}
```

Reply:
I was inverted "if" condition.